2025-03-28 16:44:22.863321+01:00 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Columna "CLIENTE_ID" no encontrada
Column "CLIENTE_ID" not found; SQL statement:
INSERT INTO LineaVentas (id, cliente_id, total, created_at, updated_at, deleted) VALUES (?, ?, ?, ?, ?, ?) [42122-232]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:514)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	at org.h2.message.DbException.get(DbException.java:223)
	at org.h2.message.DbException.get(DbException.java:199)
	at org.h2.table.Table.getColumn(Table.java:786)
	at org.h2.command.Parser.parseColumn(Parser.java:1171)
	at org.h2.command.Parser.parseColumnList(Parser.java:1156)
	at org.h2.command.Parser.parseInsert(Parser.java:1530)
	at org.h2.command.Parser.parsePrepared(Parser.java:698)
	at org.h2.command.Parser.parse(Parser.java:581)
	at org.h2.command.Parser.parse(Parser.java:556)
	at org.h2.command.Parser.prepareCommand(Parser.java:484)
	at org.h2.engine.SessionLocal.prepareLocal(SessionLocal.java:645)
	at org.h2.engine.SessionLocal.prepareCommand(SessionLocal.java:561)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1164)
	at org.h2.jdbc.JdbcPreparedStatement.<init>(JdbcPreparedStatement.java:93)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:687)
	at org.jdbi.v3.core.statement.DefaultStatementBuilder.create(DefaultStatementBuilder.java:57)
	at org.jdbi.v3.core.statement.SqlStatement.createStatement(SqlStatement.java:1834)
	at org.jdbi.v3.core.statement.SqlStatement.internalExecute(SqlStatement.java:1803)
	at org.jdbi.v3.core.result.ResultProducers.lambda$returningUpdateCount$0(ResultProducers.java:50)
	at org.jdbi.v3.core.statement.Update.execute(Update.java:69)
	at org.jdbi.v3.core.statement.Update.execute(Update.java:57)
	at org.jdbi.v3.sqlobject.statement.internal.SqlUpdateHandler.lambda$configureReturner$2(SqlUpdateHandler.java:95)
	at org.jdbi.v3.sqlobject.statement.internal.CustomizingStatementHandler.invoke(CustomizingStatementHandler.java:197)
	at org.jdbi.v3.sqlobject.statement.internal.SqlUpdateHandler.invoke(SqlUpdateHandler.java:34)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.lambda$invoke$0(ExtensionMetadata.java:345)
	at org.jdbi.v3.core.AbstractHandleSupplier.invokeInContext(AbstractHandleSupplier.java:36)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.call(ExtensionMetadata.java:363)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.invoke(ExtensionMetadata.java:346)
	at org.jdbi.v3.core.extension.ExtensionFactoryDelegate.lambda$attach$0(ExtensionFactoryDelegate.java:118)
	at org.jdbi.v3.core.internal.OnDemandExtensions.invoke(OnDemandExtensions.java:86)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$2(OnDemandExtensions.java:72)
	at org.jdbi.v3.core.Jdbi.callWithExtension(Jdbi.java:557)
	at org.jdbi.v3.core.Jdbi.withExtension(Jdbi.java:544)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$3(OnDemandExtensions.java:72)
	at jdk.proxy2/jdk.proxy2.$Proxy42.save(Unknown Source)
	at dev.joseluisgs.ventas.repository.VentasRepositoryImpl.save(VentasRepositoryImpl.kt:70)
	at dev.joseluisgs.ventas.repository.VentasRepositoryImpl.save(VentasRepositoryImpl.kt:16)
	at dev.joseluisgs.ventas.service.VentasServiceImpl.create-Zyo9ksc(VentasServiceImpl.kt:49)
	at dev.joseluisgs.CarritoApp.run(Main.kt:194)
	at dev.joseluisgs.MainKt.main(Main.kt:33)
	at dev.joseluisgs.MainKt.main(Main.kt)
2025-03-28 16:45:19.601819+01:00 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Columna "CLIENTE_ID" no encontrada
Column "CLIENTE_ID" not found; SQL statement:
INSERT INTO LineaVentas (id, cliente_id, total, created_at, updated_at, deleted) VALUES (?, ?, ?, ?, ?, ?) [42122-232]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:514)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	at org.h2.message.DbException.get(DbException.java:223)
	at org.h2.message.DbException.get(DbException.java:199)
	at org.h2.table.Table.getColumn(Table.java:786)
	at org.h2.command.Parser.parseColumn(Parser.java:1171)
	at org.h2.command.Parser.parseColumnList(Parser.java:1156)
	at org.h2.command.Parser.parseInsert(Parser.java:1530)
	at org.h2.command.Parser.parsePrepared(Parser.java:698)
	at org.h2.command.Parser.parse(Parser.java:581)
	at org.h2.command.Parser.parse(Parser.java:556)
	at org.h2.command.Parser.prepareCommand(Parser.java:484)
	at org.h2.engine.SessionLocal.prepareLocal(SessionLocal.java:645)
	at org.h2.engine.SessionLocal.prepareCommand(SessionLocal.java:561)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1164)
	at org.h2.jdbc.JdbcPreparedStatement.<init>(JdbcPreparedStatement.java:93)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:687)
	at org.jdbi.v3.core.statement.DefaultStatementBuilder.create(DefaultStatementBuilder.java:57)
	at org.jdbi.v3.core.statement.SqlStatement.createStatement(SqlStatement.java:1834)
	at org.jdbi.v3.core.statement.SqlStatement.internalExecute(SqlStatement.java:1803)
	at org.jdbi.v3.core.result.ResultProducers.lambda$returningUpdateCount$0(ResultProducers.java:50)
	at org.jdbi.v3.core.statement.Update.execute(Update.java:69)
	at org.jdbi.v3.core.statement.Update.execute(Update.java:57)
	at org.jdbi.v3.sqlobject.statement.internal.SqlUpdateHandler.lambda$configureReturner$2(SqlUpdateHandler.java:95)
	at org.jdbi.v3.sqlobject.statement.internal.CustomizingStatementHandler.invoke(CustomizingStatementHandler.java:197)
	at org.jdbi.v3.sqlobject.statement.internal.SqlUpdateHandler.invoke(SqlUpdateHandler.java:34)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.lambda$invoke$0(ExtensionMetadata.java:345)
	at org.jdbi.v3.core.AbstractHandleSupplier.invokeInContext(AbstractHandleSupplier.java:36)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.call(ExtensionMetadata.java:363)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.invoke(ExtensionMetadata.java:346)
	at org.jdbi.v3.core.extension.ExtensionFactoryDelegate.lambda$attach$0(ExtensionFactoryDelegate.java:118)
	at org.jdbi.v3.core.internal.OnDemandExtensions.invoke(OnDemandExtensions.java:86)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$2(OnDemandExtensions.java:72)
	at org.jdbi.v3.core.Jdbi.callWithExtension(Jdbi.java:557)
	at org.jdbi.v3.core.Jdbi.withExtension(Jdbi.java:544)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$3(OnDemandExtensions.java:72)
	at jdk.proxy2/jdk.proxy2.$Proxy42.save(Unknown Source)
	at dev.joseluisgs.ventas.repository.VentasRepositoryImpl.save(VentasRepositoryImpl.kt:72)
	at dev.joseluisgs.ventas.repository.VentasRepositoryImpl.save(VentasRepositoryImpl.kt:16)
	at dev.joseluisgs.ventas.service.VentasServiceImpl.create-Zyo9ksc(VentasServiceImpl.kt:49)
	at dev.joseluisgs.CarritoApp.run(Main.kt:194)
	at dev.joseluisgs.MainKt.main(Main.kt:33)
	at dev.joseluisgs.MainKt.main(Main.kt)
2025-03-28 16:46:32.907378+01:00 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Columna "TOTAL" no encontrada
Column "TOTAL" not found; SQL statement:
INSERT INTO LineaVentas (id, producto_id, total, created_at, updated_at, deleted) VALUES (?, ?, ?, ?, ?, ?) [42122-232]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:514)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	at org.h2.message.DbException.get(DbException.java:223)
	at org.h2.message.DbException.get(DbException.java:199)
	at org.h2.table.Table.getColumn(Table.java:786)
	at org.h2.command.Parser.parseColumn(Parser.java:1171)
	at org.h2.command.Parser.parseColumnList(Parser.java:1156)
	at org.h2.command.Parser.parseInsert(Parser.java:1530)
	at org.h2.command.Parser.parsePrepared(Parser.java:698)
	at org.h2.command.Parser.parse(Parser.java:581)
	at org.h2.command.Parser.parse(Parser.java:556)
	at org.h2.command.Parser.prepareCommand(Parser.java:484)
	at org.h2.engine.SessionLocal.prepareLocal(SessionLocal.java:645)
	at org.h2.engine.SessionLocal.prepareCommand(SessionLocal.java:561)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1164)
	at org.h2.jdbc.JdbcPreparedStatement.<init>(JdbcPreparedStatement.java:93)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:687)
	at org.jdbi.v3.core.statement.DefaultStatementBuilder.create(DefaultStatementBuilder.java:57)
	at org.jdbi.v3.core.statement.SqlStatement.createStatement(SqlStatement.java:1834)
	at org.jdbi.v3.core.statement.SqlStatement.internalExecute(SqlStatement.java:1803)
	at org.jdbi.v3.core.result.ResultProducers.lambda$returningUpdateCount$0(ResultProducers.java:50)
	at org.jdbi.v3.core.statement.Update.execute(Update.java:69)
	at org.jdbi.v3.core.statement.Update.execute(Update.java:57)
	at org.jdbi.v3.sqlobject.statement.internal.SqlUpdateHandler.lambda$configureReturner$2(SqlUpdateHandler.java:95)
	at org.jdbi.v3.sqlobject.statement.internal.CustomizingStatementHandler.invoke(CustomizingStatementHandler.java:197)
	at org.jdbi.v3.sqlobject.statement.internal.SqlUpdateHandler.invoke(SqlUpdateHandler.java:34)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.lambda$invoke$0(ExtensionMetadata.java:345)
	at org.jdbi.v3.core.AbstractHandleSupplier.invokeInContext(AbstractHandleSupplier.java:36)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.call(ExtensionMetadata.java:363)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.invoke(ExtensionMetadata.java:346)
	at org.jdbi.v3.core.extension.ExtensionFactoryDelegate.lambda$attach$0(ExtensionFactoryDelegate.java:118)
	at org.jdbi.v3.core.internal.OnDemandExtensions.invoke(OnDemandExtensions.java:86)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$2(OnDemandExtensions.java:72)
	at org.jdbi.v3.core.Jdbi.callWithExtension(Jdbi.java:557)
	at org.jdbi.v3.core.Jdbi.withExtension(Jdbi.java:544)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$3(OnDemandExtensions.java:72)
	at jdk.proxy2/jdk.proxy2.$Proxy42.save(Unknown Source)
	at dev.joseluisgs.ventas.repository.VentasRepositoryImpl.save(VentasRepositoryImpl.kt:72)
	at dev.joseluisgs.ventas.repository.VentasRepositoryImpl.save(VentasRepositoryImpl.kt:16)
	at dev.joseluisgs.ventas.service.VentasServiceImpl.create-Zyo9ksc(VentasServiceImpl.kt:49)
	at dev.joseluisgs.CarritoApp.run(Main.kt:194)
	at dev.joseluisgs.MainKt.main(Main.kt:33)
	at dev.joseluisgs.MainKt.main(Main.kt)
2025-03-28 16:48:26.186557+01:00 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Columna "DELETED" no encontrada
Column "DELETED" not found; SQL statement:
INSERT INTO LineaVentas (id, producto_id, cantidad, precio, created_at, updated_at, deleted) VALUES (?, ?, ?, ?, ?, ?, ?) [42122-232]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:514)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	at org.h2.message.DbException.get(DbException.java:223)
	at org.h2.message.DbException.get(DbException.java:199)
	at org.h2.table.Table.getColumn(Table.java:786)
	at org.h2.command.Parser.parseColumn(Parser.java:1171)
	at org.h2.command.Parser.parseColumnList(Parser.java:1156)
	at org.h2.command.Parser.parseInsert(Parser.java:1530)
	at org.h2.command.Parser.parsePrepared(Parser.java:698)
	at org.h2.command.Parser.parse(Parser.java:581)
	at org.h2.command.Parser.parse(Parser.java:556)
	at org.h2.command.Parser.prepareCommand(Parser.java:484)
	at org.h2.engine.SessionLocal.prepareLocal(SessionLocal.java:645)
	at org.h2.engine.SessionLocal.prepareCommand(SessionLocal.java:561)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1164)
	at org.h2.jdbc.JdbcPreparedStatement.<init>(JdbcPreparedStatement.java:93)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:687)
	at org.jdbi.v3.core.statement.DefaultStatementBuilder.create(DefaultStatementBuilder.java:57)
	at org.jdbi.v3.core.statement.SqlStatement.createStatement(SqlStatement.java:1834)
	at org.jdbi.v3.core.statement.SqlStatement.internalExecute(SqlStatement.java:1803)
	at org.jdbi.v3.core.result.ResultProducers.lambda$returningUpdateCount$0(ResultProducers.java:50)
	at org.jdbi.v3.core.statement.Update.execute(Update.java:69)
	at org.jdbi.v3.core.statement.Update.execute(Update.java:57)
	at org.jdbi.v3.sqlobject.statement.internal.SqlUpdateHandler.lambda$configureReturner$2(SqlUpdateHandler.java:95)
	at org.jdbi.v3.sqlobject.statement.internal.CustomizingStatementHandler.invoke(CustomizingStatementHandler.java:197)
	at org.jdbi.v3.sqlobject.statement.internal.SqlUpdateHandler.invoke(SqlUpdateHandler.java:34)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.lambda$invoke$0(ExtensionMetadata.java:345)
	at org.jdbi.v3.core.AbstractHandleSupplier.invokeInContext(AbstractHandleSupplier.java:36)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.call(ExtensionMetadata.java:363)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.invoke(ExtensionMetadata.java:346)
	at org.jdbi.v3.core.extension.ExtensionFactoryDelegate.lambda$attach$0(ExtensionFactoryDelegate.java:118)
	at org.jdbi.v3.core.internal.OnDemandExtensions.invoke(OnDemandExtensions.java:86)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$2(OnDemandExtensions.java:72)
	at org.jdbi.v3.core.Jdbi.callWithExtension(Jdbi.java:557)
	at org.jdbi.v3.core.Jdbi.withExtension(Jdbi.java:544)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$3(OnDemandExtensions.java:72)
	at jdk.proxy2/jdk.proxy2.$Proxy42.save(Unknown Source)
	at dev.joseluisgs.ventas.repository.VentasRepositoryImpl.save(VentasRepositoryImpl.kt:72)
	at dev.joseluisgs.ventas.repository.VentasRepositoryImpl.save(VentasRepositoryImpl.kt:16)
	at dev.joseluisgs.ventas.service.VentasServiceImpl.create-Zyo9ksc(VentasServiceImpl.kt:49)
	at dev.joseluisgs.CarritoApp.run(Main.kt:194)
	at dev.joseluisgs.MainKt.main(Main.kt:33)
	at dev.joseluisgs.MainKt.main(Main.kt)
2025-03-28 16:49:28.475993+01:00 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Columna "DELETED" no encontrada
Column "DELETED" not found; SQL statement:
INSERT INTO LineaVentas (id, producto_id, cantidad, precio, created_at, updated_at, deleted) VALUES (?, ?, ?, ?, ?, ?) [42122-232]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:514)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	at org.h2.message.DbException.get(DbException.java:223)
	at org.h2.message.DbException.get(DbException.java:199)
	at org.h2.table.Table.getColumn(Table.java:786)
	at org.h2.command.Parser.parseColumn(Parser.java:1171)
	at org.h2.command.Parser.parseColumnList(Parser.java:1156)
	at org.h2.command.Parser.parseInsert(Parser.java:1530)
	at org.h2.command.Parser.parsePrepared(Parser.java:698)
	at org.h2.command.Parser.parse(Parser.java:581)
	at org.h2.command.Parser.parse(Parser.java:556)
	at org.h2.command.Parser.prepareCommand(Parser.java:484)
	at org.h2.engine.SessionLocal.prepareLocal(SessionLocal.java:645)
	at org.h2.engine.SessionLocal.prepareCommand(SessionLocal.java:561)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1164)
	at org.h2.jdbc.JdbcPreparedStatement.<init>(JdbcPreparedStatement.java:93)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:687)
	at org.jdbi.v3.core.statement.DefaultStatementBuilder.create(DefaultStatementBuilder.java:57)
	at org.jdbi.v3.core.statement.SqlStatement.createStatement(SqlStatement.java:1834)
	at org.jdbi.v3.core.statement.SqlStatement.internalExecute(SqlStatement.java:1803)
	at org.jdbi.v3.core.result.ResultProducers.lambda$returningUpdateCount$0(ResultProducers.java:50)
	at org.jdbi.v3.core.statement.Update.execute(Update.java:69)
	at org.jdbi.v3.core.statement.Update.execute(Update.java:57)
	at org.jdbi.v3.sqlobject.statement.internal.SqlUpdateHandler.lambda$configureReturner$2(SqlUpdateHandler.java:95)
	at org.jdbi.v3.sqlobject.statement.internal.CustomizingStatementHandler.invoke(CustomizingStatementHandler.java:197)
	at org.jdbi.v3.sqlobject.statement.internal.SqlUpdateHandler.invoke(SqlUpdateHandler.java:34)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.lambda$invoke$0(ExtensionMetadata.java:345)
	at org.jdbi.v3.core.AbstractHandleSupplier.invokeInContext(AbstractHandleSupplier.java:36)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.call(ExtensionMetadata.java:363)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.invoke(ExtensionMetadata.java:346)
	at org.jdbi.v3.core.extension.ExtensionFactoryDelegate.lambda$attach$0(ExtensionFactoryDelegate.java:118)
	at org.jdbi.v3.core.internal.OnDemandExtensions.invoke(OnDemandExtensions.java:86)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$2(OnDemandExtensions.java:72)
	at org.jdbi.v3.core.Jdbi.callWithExtension(Jdbi.java:557)
	at org.jdbi.v3.core.Jdbi.withExtension(Jdbi.java:544)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$3(OnDemandExtensions.java:72)
	at jdk.proxy2/jdk.proxy2.$Proxy42.save(Unknown Source)
	at dev.joseluisgs.ventas.repository.VentasRepositoryImpl.save(VentasRepositoryImpl.kt:72)
	at dev.joseluisgs.ventas.repository.VentasRepositoryImpl.save(VentasRepositoryImpl.kt:16)
	at dev.joseluisgs.ventas.service.VentasServiceImpl.create-Zyo9ksc(VentasServiceImpl.kt:49)
	at dev.joseluisgs.CarritoApp.run(Main.kt:194)
	at dev.joseluisgs.MainKt.main(Main.kt:33)
	at dev.joseluisgs.MainKt.main(Main.kt)
2025-04-02 19:23:15.359118+02:00 database: flush
org.h2.message.DbException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:407)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:122)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:1546)
	at org.h2.mvstore.MVStore.panic(MVStore.java:371)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:291)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2035)
	at org.h2.mvstore.db.Store.<init>(Store.java:133)
	at org.h2.engine.Database.<init>(Database.java:326)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:344)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.Driver.connect(Driver.java:59)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:683)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:253)
	at org.jdbi.v3.core.Jdbi.lambda$create$0(Jdbi.java:122)
	at org.jdbi.v3.core.Jdbi.open(Jdbi.java:366)
	at org.jdbi.v3.core.LazyHandleSupplier.createHandle(LazyHandleSupplier.java:50)
	at org.jdbi.v3.core.internal.MemoizingSupplier.init(MemoizingSupplier.java:72)
	at org.jdbi.v3.core.internal.MemoizingSupplier.get(MemoizingSupplier.java:49)
	at org.jdbi.v3.core.LazyHandleSupplier.getHandle(LazyHandleSupplier.java:45)
	at org.jdbi.v3.sqlobject.statement.internal.CustomizingStatementHandler.invoke(CustomizingStatementHandler.java:186)
	at org.jdbi.v3.sqlobject.statement.internal.SqlQueryHandler.invoke(SqlQueryHandler.java:27)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.lambda$invoke$0(ExtensionMetadata.java:345)
	at org.jdbi.v3.core.AbstractHandleSupplier.invokeInContext(AbstractHandleSupplier.java:36)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.call(ExtensionMetadata.java:363)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.invoke(ExtensionMetadata.java:346)
	at org.jdbi.v3.core.extension.ExtensionFactoryDelegate.lambda$attach$0(ExtensionFactoryDelegate.java:118)
	at org.jdbi.v3.core.internal.OnDemandExtensions.invoke(OnDemandExtensions.java:86)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$2(OnDemandExtensions.java:72)
	at org.jdbi.v3.core.Jdbi.callWithExtension(Jdbi.java:557)
	at org.jdbi.v3.core.Jdbi.withExtension(Jdbi.java:544)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$3(OnDemandExtensions.java:72)
	at jdk.proxy2/jdk.proxy2.$Proxy27.findAll(Unknown Source)
	at dev.joseluisgs.productos.repository.ProductosRepositoryImpl.findAll(ProductosRepositoryImpl.kt:19)
	at dev.joseluisgs.productos.service.ProductosServiceImpl.getAll--HJRILA(ProductosServiceImpl.kt:29)
	at dev.joseluisgs.CarritoApp.run(Main.kt:51)
	at dev.joseluisgs.MainKt.main(Main.kt:36)
	at dev.joseluisgs.MainKt.main(Main.kt)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 41 more
Caused by: org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.SingleFileStore.lockFileChannel(SingleFileStore.java:143)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:117)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:81)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:286)
	... 35 more
2025-04-02 19:23:15.382125+02:00 database: flush
org.h2.message.DbException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:407)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:122)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:1546)
	at org.h2.mvstore.MVStore.panic(MVStore.java:371)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:291)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2035)
	at org.h2.mvstore.db.Store.<init>(Store.java:133)
	at org.h2.engine.Database.<init>(Database.java:326)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:344)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.Driver.connect(Driver.java:59)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:683)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:253)
	at org.jdbi.v3.core.Jdbi.lambda$create$0(Jdbi.java:122)
	at org.jdbi.v3.core.Jdbi.open(Jdbi.java:366)
	at org.jdbi.v3.core.LazyHandleSupplier.createHandle(LazyHandleSupplier.java:50)
	at org.jdbi.v3.core.internal.MemoizingSupplier.init(MemoizingSupplier.java:72)
	at org.jdbi.v3.core.internal.MemoizingSupplier.get(MemoizingSupplier.java:49)
	at org.jdbi.v3.core.LazyHandleSupplier.getHandle(LazyHandleSupplier.java:45)
	at org.jdbi.v3.sqlobject.statement.internal.CustomizingStatementHandler.invoke(CustomizingStatementHandler.java:186)
	at org.jdbi.v3.sqlobject.statement.internal.SqlQueryHandler.invoke(SqlQueryHandler.java:27)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.lambda$invoke$0(ExtensionMetadata.java:345)
	at org.jdbi.v3.core.AbstractHandleSupplier.invokeInContext(AbstractHandleSupplier.java:36)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.call(ExtensionMetadata.java:363)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.invoke(ExtensionMetadata.java:346)
	at org.jdbi.v3.core.extension.ExtensionFactoryDelegate.lambda$attach$0(ExtensionFactoryDelegate.java:118)
	at org.jdbi.v3.core.internal.OnDemandExtensions.invoke(OnDemandExtensions.java:86)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$2(OnDemandExtensions.java:72)
	at org.jdbi.v3.core.Jdbi.callWithExtension(Jdbi.java:557)
	at org.jdbi.v3.core.Jdbi.withExtension(Jdbi.java:544)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$3(OnDemandExtensions.java:72)
	at jdk.proxy2/jdk.proxy2.$Proxy27.findById(Unknown Source)
	at dev.joseluisgs.productos.repository.ProductosRepositoryImpl.findById(ProductosRepositoryImpl.kt:24)
	at dev.joseluisgs.productos.repository.ProductosRepositoryImpl.findById(ProductosRepositoryImpl.kt:11)
	at dev.joseluisgs.productos.service.ProductosServiceImpl.getById-Zyo9ksc(ProductosServiceImpl.kt:42)
	at dev.joseluisgs.CarritoApp.run(Main.kt:58)
	at dev.joseluisgs.MainKt.main(Main.kt:36)
	at dev.joseluisgs.MainKt.main(Main.kt)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 42 more
Caused by: org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.SingleFileStore.lockFileChannel(SingleFileStore.java:143)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:117)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:81)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:286)
	... 36 more
2025-04-02 19:23:15.399118+02:00 database: flush
org.h2.message.DbException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:407)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:122)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:1546)
	at org.h2.mvstore.MVStore.panic(MVStore.java:371)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:291)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2035)
	at org.h2.mvstore.db.Store.<init>(Store.java:133)
	at org.h2.engine.Database.<init>(Database.java:326)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:344)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.Driver.connect(Driver.java:59)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:683)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:253)
	at org.jdbi.v3.core.Jdbi.lambda$create$0(Jdbi.java:122)
	at org.jdbi.v3.core.Jdbi.open(Jdbi.java:366)
	at org.jdbi.v3.core.LazyHandleSupplier.createHandle(LazyHandleSupplier.java:50)
	at org.jdbi.v3.core.internal.MemoizingSupplier.init(MemoizingSupplier.java:72)
	at org.jdbi.v3.core.internal.MemoizingSupplier.get(MemoizingSupplier.java:49)
	at org.jdbi.v3.core.LazyHandleSupplier.getHandle(LazyHandleSupplier.java:45)
	at org.jdbi.v3.sqlobject.statement.internal.CustomizingStatementHandler.invoke(CustomizingStatementHandler.java:186)
	at org.jdbi.v3.sqlobject.statement.internal.SqlQueryHandler.invoke(SqlQueryHandler.java:27)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.lambda$invoke$0(ExtensionMetadata.java:345)
	at org.jdbi.v3.core.AbstractHandleSupplier.invokeInContext(AbstractHandleSupplier.java:36)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.call(ExtensionMetadata.java:363)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.invoke(ExtensionMetadata.java:346)
	at org.jdbi.v3.core.extension.ExtensionFactoryDelegate.lambda$attach$0(ExtensionFactoryDelegate.java:118)
	at org.jdbi.v3.core.internal.OnDemandExtensions.invoke(OnDemandExtensions.java:86)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$2(OnDemandExtensions.java:72)
	at org.jdbi.v3.core.Jdbi.callWithExtension(Jdbi.java:557)
	at org.jdbi.v3.core.Jdbi.withExtension(Jdbi.java:544)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$3(OnDemandExtensions.java:72)
	at jdk.proxy2/jdk.proxy2.$Proxy27.findById(Unknown Source)
	at dev.joseluisgs.productos.repository.ProductosRepositoryImpl.findById(ProductosRepositoryImpl.kt:24)
	at dev.joseluisgs.productos.repository.ProductosRepositoryImpl.findById(ProductosRepositoryImpl.kt:11)
	at dev.joseluisgs.productos.service.ProductosServiceImpl.getById-Zyo9ksc(ProductosServiceImpl.kt:42)
	at dev.joseluisgs.CarritoApp.run(Main.kt:64)
	at dev.joseluisgs.MainKt.main(Main.kt:36)
	at dev.joseluisgs.MainKt.main(Main.kt)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 42 more
Caused by: org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.SingleFileStore.lockFileChannel(SingleFileStore.java:143)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:117)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:81)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:286)
	... 36 more
2025-04-02 19:23:15.422119+02:00 database: flush
org.h2.message.DbException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:407)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:122)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:1546)
	at org.h2.mvstore.MVStore.panic(MVStore.java:371)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:291)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2035)
	at org.h2.mvstore.db.Store.<init>(Store.java:133)
	at org.h2.engine.Database.<init>(Database.java:326)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:344)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.Driver.connect(Driver.java:59)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:683)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:253)
	at org.jdbi.v3.core.Jdbi.lambda$create$0(Jdbi.java:122)
	at org.jdbi.v3.core.Jdbi.open(Jdbi.java:366)
	at org.jdbi.v3.core.LazyHandleSupplier.createHandle(LazyHandleSupplier.java:50)
	at org.jdbi.v3.core.internal.MemoizingSupplier.init(MemoizingSupplier.java:72)
	at org.jdbi.v3.core.internal.MemoizingSupplier.get(MemoizingSupplier.java:49)
	at org.jdbi.v3.core.LazyHandleSupplier.getHandle(LazyHandleSupplier.java:45)
	at org.jdbi.v3.sqlobject.statement.internal.CustomizingStatementHandler.invoke(CustomizingStatementHandler.java:186)
	at org.jdbi.v3.sqlobject.statement.internal.SqlUpdateHandler.invoke(SqlUpdateHandler.java:34)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.lambda$invoke$0(ExtensionMetadata.java:345)
	at org.jdbi.v3.core.AbstractHandleSupplier.invokeInContext(AbstractHandleSupplier.java:36)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.call(ExtensionMetadata.java:363)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.invoke(ExtensionMetadata.java:346)
	at org.jdbi.v3.core.extension.ExtensionFactoryDelegate.lambda$attach$0(ExtensionFactoryDelegate.java:118)
	at org.jdbi.v3.core.internal.OnDemandExtensions.invoke(OnDemandExtensions.java:86)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$2(OnDemandExtensions.java:72)
	at org.jdbi.v3.core.Jdbi.callWithExtension(Jdbi.java:557)
	at org.jdbi.v3.core.Jdbi.withExtension(Jdbi.java:544)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$3(OnDemandExtensions.java:72)
	at jdk.proxy2/jdk.proxy2.$Proxy27.save(Unknown Source)
	at dev.joseluisgs.productos.repository.ProductosRepositoryImpl.save(ProductosRepositoryImpl.kt:31)
	at dev.joseluisgs.productos.repository.ProductosRepositoryImpl.save(ProductosRepositoryImpl.kt:11)
	at dev.joseluisgs.productos.service.ProductosServiceImpl.create-Zyo9ksc(ProductosServiceImpl.kt:56)
	at dev.joseluisgs.CarritoApp.run(Main.kt:76)
	at dev.joseluisgs.MainKt.main(Main.kt:36)
	at dev.joseluisgs.MainKt.main(Main.kt)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 42 more
Caused by: org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.SingleFileStore.lockFileChannel(SingleFileStore.java:143)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:117)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:81)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:286)
	... 36 more
2025-04-02 19:23:15.438119+02:00 database: flush
org.h2.message.DbException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:407)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:122)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:1546)
	at org.h2.mvstore.MVStore.panic(MVStore.java:371)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:291)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2035)
	at org.h2.mvstore.db.Store.<init>(Store.java:133)
	at org.h2.engine.Database.<init>(Database.java:326)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:344)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.Driver.connect(Driver.java:59)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:683)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:253)
	at org.jdbi.v3.core.Jdbi.lambda$create$0(Jdbi.java:122)
	at org.jdbi.v3.core.Jdbi.open(Jdbi.java:366)
	at org.jdbi.v3.core.LazyHandleSupplier.createHandle(LazyHandleSupplier.java:50)
	at org.jdbi.v3.core.internal.MemoizingSupplier.init(MemoizingSupplier.java:72)
	at org.jdbi.v3.core.internal.MemoizingSupplier.get(MemoizingSupplier.java:49)
	at org.jdbi.v3.core.LazyHandleSupplier.getHandle(LazyHandleSupplier.java:45)
	at org.jdbi.v3.sqlobject.statement.internal.CustomizingStatementHandler.invoke(CustomizingStatementHandler.java:186)
	at org.jdbi.v3.sqlobject.statement.internal.SqlQueryHandler.invoke(SqlQueryHandler.java:27)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.lambda$invoke$0(ExtensionMetadata.java:345)
	at org.jdbi.v3.core.AbstractHandleSupplier.invokeInContext(AbstractHandleSupplier.java:36)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.call(ExtensionMetadata.java:363)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.invoke(ExtensionMetadata.java:346)
	at org.jdbi.v3.core.extension.ExtensionFactoryDelegate.lambda$attach$0(ExtensionFactoryDelegate.java:118)
	at org.jdbi.v3.core.internal.OnDemandExtensions.invoke(OnDemandExtensions.java:86)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$2(OnDemandExtensions.java:72)
	at org.jdbi.v3.core.Jdbi.callWithExtension(Jdbi.java:557)
	at org.jdbi.v3.core.Jdbi.withExtension(Jdbi.java:544)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$3(OnDemandExtensions.java:72)
	at jdk.proxy2/jdk.proxy2.$Proxy27.findById(Unknown Source)
	at dev.joseluisgs.productos.repository.ProductosRepositoryImpl.findById(ProductosRepositoryImpl.kt:24)
	at dev.joseluisgs.productos.repository.ProductosRepositoryImpl.update(ProductosRepositoryImpl.kt:38)
	at dev.joseluisgs.productos.repository.ProductosRepositoryImpl.update(ProductosRepositoryImpl.kt:11)
	at dev.joseluisgs.productos.service.ProductosServiceImpl.update-T3BZVFY(ProductosServiceImpl.kt:68)
	at dev.joseluisgs.CarritoApp.run(Main.kt:94)
	at dev.joseluisgs.MainKt.main(Main.kt:36)
	at dev.joseluisgs.MainKt.main(Main.kt)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 43 more
Caused by: org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.SingleFileStore.lockFileChannel(SingleFileStore.java:143)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:117)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:81)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:286)
	... 37 more
2025-04-02 19:23:15.452119+02:00 database: flush
org.h2.message.DbException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:407)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:122)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:1546)
	at org.h2.mvstore.MVStore.panic(MVStore.java:371)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:291)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2035)
	at org.h2.mvstore.db.Store.<init>(Store.java:133)
	at org.h2.engine.Database.<init>(Database.java:326)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:344)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.Driver.connect(Driver.java:59)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:683)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:253)
	at org.jdbi.v3.core.Jdbi.lambda$create$0(Jdbi.java:122)
	at org.jdbi.v3.core.Jdbi.open(Jdbi.java:366)
	at org.jdbi.v3.core.LazyHandleSupplier.createHandle(LazyHandleSupplier.java:50)
	at org.jdbi.v3.core.internal.MemoizingSupplier.init(MemoizingSupplier.java:72)
	at org.jdbi.v3.core.internal.MemoizingSupplier.get(MemoizingSupplier.java:49)
	at org.jdbi.v3.core.LazyHandleSupplier.getHandle(LazyHandleSupplier.java:45)
	at org.jdbi.v3.sqlobject.statement.internal.CustomizingStatementHandler.invoke(CustomizingStatementHandler.java:186)
	at org.jdbi.v3.sqlobject.statement.internal.SqlQueryHandler.invoke(SqlQueryHandler.java:27)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.lambda$invoke$0(ExtensionMetadata.java:345)
	at org.jdbi.v3.core.AbstractHandleSupplier.invokeInContext(AbstractHandleSupplier.java:36)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.call(ExtensionMetadata.java:363)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.invoke(ExtensionMetadata.java:346)
	at org.jdbi.v3.core.extension.ExtensionFactoryDelegate.lambda$attach$0(ExtensionFactoryDelegate.java:118)
	at org.jdbi.v3.core.internal.OnDemandExtensions.invoke(OnDemandExtensions.java:86)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$2(OnDemandExtensions.java:72)
	at org.jdbi.v3.core.Jdbi.callWithExtension(Jdbi.java:557)
	at org.jdbi.v3.core.Jdbi.withExtension(Jdbi.java:544)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$3(OnDemandExtensions.java:72)
	at jdk.proxy2/jdk.proxy2.$Proxy27.findById(Unknown Source)
	at dev.joseluisgs.productos.repository.ProductosRepositoryImpl.findById(ProductosRepositoryImpl.kt:24)
	at dev.joseluisgs.productos.repository.ProductosRepositoryImpl.findById(ProductosRepositoryImpl.kt:11)
	at dev.joseluisgs.productos.service.ProductosServiceImpl.getById-Zyo9ksc(ProductosServiceImpl.kt:42)
	at dev.joseluisgs.CarritoApp.run(Main.kt:102)
	at dev.joseluisgs.MainKt.main(Main.kt:36)
	at dev.joseluisgs.MainKt.main(Main.kt)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 42 more
Caused by: org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.SingleFileStore.lockFileChannel(SingleFileStore.java:143)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:117)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:81)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:286)
	... 36 more
2025-04-02 19:23:15.466118+02:00 database: flush
org.h2.message.DbException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:407)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:122)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:1546)
	at org.h2.mvstore.MVStore.panic(MVStore.java:371)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:291)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2035)
	at org.h2.mvstore.db.Store.<init>(Store.java:133)
	at org.h2.engine.Database.<init>(Database.java:326)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:344)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.Driver.connect(Driver.java:59)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:683)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:253)
	at org.jdbi.v3.core.Jdbi.lambda$create$0(Jdbi.java:122)
	at org.jdbi.v3.core.Jdbi.open(Jdbi.java:366)
	at org.jdbi.v3.core.LazyHandleSupplier.createHandle(LazyHandleSupplier.java:50)
	at org.jdbi.v3.core.internal.MemoizingSupplier.init(MemoizingSupplier.java:72)
	at org.jdbi.v3.core.internal.MemoizingSupplier.get(MemoizingSupplier.java:49)
	at org.jdbi.v3.core.LazyHandleSupplier.getHandle(LazyHandleSupplier.java:45)
	at org.jdbi.v3.sqlobject.statement.internal.CustomizingStatementHandler.invoke(CustomizingStatementHandler.java:186)
	at org.jdbi.v3.sqlobject.statement.internal.SqlQueryHandler.invoke(SqlQueryHandler.java:27)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.lambda$invoke$0(ExtensionMetadata.java:345)
	at org.jdbi.v3.core.AbstractHandleSupplier.invokeInContext(AbstractHandleSupplier.java:36)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.call(ExtensionMetadata.java:363)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.invoke(ExtensionMetadata.java:346)
	at org.jdbi.v3.core.extension.ExtensionFactoryDelegate.lambda$attach$0(ExtensionFactoryDelegate.java:118)
	at org.jdbi.v3.core.internal.OnDemandExtensions.invoke(OnDemandExtensions.java:86)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$2(OnDemandExtensions.java:72)
	at org.jdbi.v3.core.Jdbi.callWithExtension(Jdbi.java:557)
	at org.jdbi.v3.core.Jdbi.withExtension(Jdbi.java:544)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$3(OnDemandExtensions.java:72)
	at jdk.proxy2/jdk.proxy2.$Proxy27.findAll(Unknown Source)
	at dev.joseluisgs.productos.repository.ProductosRepositoryImpl.findAll(ProductosRepositoryImpl.kt:19)
	at dev.joseluisgs.productos.service.ProductosServiceImpl.getAll--HJRILA(ProductosServiceImpl.kt:29)
	at dev.joseluisgs.CarritoApp.run(Main.kt:108)
	at dev.joseluisgs.MainKt.main(Main.kt:36)
	at dev.joseluisgs.MainKt.main(Main.kt)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 41 more
Caused by: org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.SingleFileStore.lockFileChannel(SingleFileStore.java:143)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:117)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:81)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:286)
	... 35 more
2025-04-02 19:23:15.481119+02:00 database: flush
org.h2.message.DbException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:407)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:122)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:1546)
	at org.h2.mvstore.MVStore.panic(MVStore.java:371)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:291)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2035)
	at org.h2.mvstore.db.Store.<init>(Store.java:133)
	at org.h2.engine.Database.<init>(Database.java:326)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:344)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.Driver.connect(Driver.java:59)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:683)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:253)
	at org.jdbi.v3.core.Jdbi.lambda$create$0(Jdbi.java:122)
	at org.jdbi.v3.core.Jdbi.open(Jdbi.java:366)
	at org.jdbi.v3.core.LazyHandleSupplier.createHandle(LazyHandleSupplier.java:50)
	at org.jdbi.v3.core.internal.MemoizingSupplier.init(MemoizingSupplier.java:72)
	at org.jdbi.v3.core.internal.MemoizingSupplier.get(MemoizingSupplier.java:49)
	at org.jdbi.v3.core.LazyHandleSupplier.getHandle(LazyHandleSupplier.java:45)
	at org.jdbi.v3.sqlobject.statement.internal.CustomizingStatementHandler.invoke(CustomizingStatementHandler.java:186)
	at org.jdbi.v3.sqlobject.statement.internal.SqlQueryHandler.invoke(SqlQueryHandler.java:27)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.lambda$invoke$0(ExtensionMetadata.java:345)
	at org.jdbi.v3.core.AbstractHandleSupplier.invokeInContext(AbstractHandleSupplier.java:36)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.call(ExtensionMetadata.java:363)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.invoke(ExtensionMetadata.java:346)
	at org.jdbi.v3.core.extension.ExtensionFactoryDelegate.lambda$attach$0(ExtensionFactoryDelegate.java:118)
	at org.jdbi.v3.core.internal.OnDemandExtensions.invoke(OnDemandExtensions.java:86)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$2(OnDemandExtensions.java:72)
	at org.jdbi.v3.core.Jdbi.callWithExtension(Jdbi.java:557)
	at org.jdbi.v3.core.Jdbi.withExtension(Jdbi.java:544)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$3(OnDemandExtensions.java:72)
	at jdk.proxy2/jdk.proxy2.$Proxy27.findById(Unknown Source)
	at dev.joseluisgs.productos.repository.ProductosRepositoryImpl.findById(ProductosRepositoryImpl.kt:24)
	at dev.joseluisgs.productos.repository.ProductosRepositoryImpl.deleteLogico(ProductosRepositoryImpl.kt:53)
	at dev.joseluisgs.productos.service.ProductosServiceImpl.delete-Zyo9ksc(ProductosServiceImpl.kt:83)
	at dev.joseluisgs.CarritoApp.run(Main.kt:115)
	at dev.joseluisgs.MainKt.main(Main.kt:36)
	at dev.joseluisgs.MainKt.main(Main.kt)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 42 more
Caused by: org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.SingleFileStore.lockFileChannel(SingleFileStore.java:143)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:117)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:81)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:286)
	... 36 more
2025-04-02 19:23:15.499131+02:00 database: flush
org.h2.message.DbException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:407)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:122)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:1546)
	at org.h2.mvstore.MVStore.panic(MVStore.java:371)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:291)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2035)
	at org.h2.mvstore.db.Store.<init>(Store.java:133)
	at org.h2.engine.Database.<init>(Database.java:326)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:344)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.Driver.connect(Driver.java:59)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:683)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:253)
	at org.jdbi.v3.core.Jdbi.lambda$create$0(Jdbi.java:122)
	at org.jdbi.v3.core.Jdbi.open(Jdbi.java:366)
	at org.jdbi.v3.core.LazyHandleSupplier.createHandle(LazyHandleSupplier.java:50)
	at org.jdbi.v3.core.internal.MemoizingSupplier.init(MemoizingSupplier.java:72)
	at org.jdbi.v3.core.internal.MemoizingSupplier.get(MemoizingSupplier.java:49)
	at org.jdbi.v3.core.LazyHandleSupplier.getHandle(LazyHandleSupplier.java:45)
	at org.jdbi.v3.sqlobject.statement.internal.CustomizingStatementHandler.invoke(CustomizingStatementHandler.java:186)
	at org.jdbi.v3.sqlobject.statement.internal.SqlQueryHandler.invoke(SqlQueryHandler.java:27)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.lambda$invoke$0(ExtensionMetadata.java:345)
	at org.jdbi.v3.core.AbstractHandleSupplier.invokeInContext(AbstractHandleSupplier.java:36)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.call(ExtensionMetadata.java:363)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.invoke(ExtensionMetadata.java:346)
	at org.jdbi.v3.core.extension.ExtensionFactoryDelegate.lambda$attach$0(ExtensionFactoryDelegate.java:118)
	at org.jdbi.v3.core.internal.OnDemandExtensions.invoke(OnDemandExtensions.java:86)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$2(OnDemandExtensions.java:72)
	at org.jdbi.v3.core.Jdbi.callWithExtension(Jdbi.java:557)
	at org.jdbi.v3.core.Jdbi.withExtension(Jdbi.java:544)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$3(OnDemandExtensions.java:72)
	at jdk.proxy2/jdk.proxy2.$Proxy27.findById(Unknown Source)
	at dev.joseluisgs.productos.repository.ProductosRepositoryImpl.findById(ProductosRepositoryImpl.kt:24)
	at dev.joseluisgs.productos.repository.ProductosRepositoryImpl.deleteLogico(ProductosRepositoryImpl.kt:53)
	at dev.joseluisgs.productos.service.ProductosServiceImpl.delete-Zyo9ksc(ProductosServiceImpl.kt:83)
	at dev.joseluisgs.CarritoApp.run(Main.kt:121)
	at dev.joseluisgs.MainKt.main(Main.kt:36)
	at dev.joseluisgs.MainKt.main(Main.kt)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 42 more
Caused by: org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.SingleFileStore.lockFileChannel(SingleFileStore.java:143)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:117)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:81)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:286)
	... 36 more
2025-04-02 19:23:15.516662+02:00 database: flush
org.h2.message.DbException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:407)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:122)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:1546)
	at org.h2.mvstore.MVStore.panic(MVStore.java:371)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:291)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2035)
	at org.h2.mvstore.db.Store.<init>(Store.java:133)
	at org.h2.engine.Database.<init>(Database.java:326)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:344)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.Driver.connect(Driver.java:59)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:683)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:253)
	at org.jdbi.v3.core.Jdbi.lambda$create$0(Jdbi.java:122)
	at org.jdbi.v3.core.Jdbi.open(Jdbi.java:366)
	at org.jdbi.v3.core.LazyHandleSupplier.createHandle(LazyHandleSupplier.java:50)
	at org.jdbi.v3.core.internal.MemoizingSupplier.init(MemoizingSupplier.java:72)
	at org.jdbi.v3.core.internal.MemoizingSupplier.get(MemoizingSupplier.java:49)
	at org.jdbi.v3.core.LazyHandleSupplier.getHandle(LazyHandleSupplier.java:45)
	at org.jdbi.v3.sqlobject.statement.internal.CustomizingStatementHandler.invoke(CustomizingStatementHandler.java:186)
	at org.jdbi.v3.sqlobject.statement.internal.SqlQueryHandler.invoke(SqlQueryHandler.java:27)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.lambda$invoke$0(ExtensionMetadata.java:345)
	at org.jdbi.v3.core.AbstractHandleSupplier.invokeInContext(AbstractHandleSupplier.java:36)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.call(ExtensionMetadata.java:363)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.invoke(ExtensionMetadata.java:346)
	at org.jdbi.v3.core.extension.ExtensionFactoryDelegate.lambda$attach$0(ExtensionFactoryDelegate.java:118)
	at org.jdbi.v3.core.internal.OnDemandExtensions.invoke(OnDemandExtensions.java:86)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$2(OnDemandExtensions.java:72)
	at org.jdbi.v3.core.Jdbi.callWithExtension(Jdbi.java:557)
	at org.jdbi.v3.core.Jdbi.withExtension(Jdbi.java:544)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$3(OnDemandExtensions.java:72)
	at jdk.proxy2/jdk.proxy2.$Proxy27.findById(Unknown Source)
	at dev.joseluisgs.productos.repository.ProductosRepositoryImpl.findById(ProductosRepositoryImpl.kt:24)
	at dev.joseluisgs.productos.repository.ProductosRepositoryImpl.findById(ProductosRepositoryImpl.kt:11)
	at dev.joseluisgs.productos.service.ProductosServiceImpl.getById-Zyo9ksc(ProductosServiceImpl.kt:42)
	at dev.joseluisgs.CarritoApp.run(Main.kt:128)
	at dev.joseluisgs.MainKt.main(Main.kt:36)
	at dev.joseluisgs.MainKt.main(Main.kt)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 42 more
Caused by: org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.SingleFileStore.lockFileChannel(SingleFileStore.java:143)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:117)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:81)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:286)
	... 36 more
2025-04-02 19:23:15.534661+02:00 database: flush
org.h2.message.DbException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:407)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:122)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:1546)
	at org.h2.mvstore.MVStore.panic(MVStore.java:371)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:291)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2035)
	at org.h2.mvstore.db.Store.<init>(Store.java:133)
	at org.h2.engine.Database.<init>(Database.java:326)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:344)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.Driver.connect(Driver.java:59)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:683)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:253)
	at org.jdbi.v3.core.Jdbi.lambda$create$0(Jdbi.java:122)
	at org.jdbi.v3.core.Jdbi.open(Jdbi.java:366)
	at org.jdbi.v3.core.LazyHandleSupplier.createHandle(LazyHandleSupplier.java:50)
	at org.jdbi.v3.core.internal.MemoizingSupplier.init(MemoizingSupplier.java:72)
	at org.jdbi.v3.core.internal.MemoizingSupplier.get(MemoizingSupplier.java:49)
	at org.jdbi.v3.core.LazyHandleSupplier.getHandle(LazyHandleSupplier.java:45)
	at org.jdbi.v3.sqlobject.statement.internal.CustomizingStatementHandler.invoke(CustomizingStatementHandler.java:186)
	at org.jdbi.v3.sqlobject.statement.internal.SqlQueryHandler.invoke(SqlQueryHandler.java:27)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.lambda$invoke$0(ExtensionMetadata.java:345)
	at org.jdbi.v3.core.AbstractHandleSupplier.invokeInContext(AbstractHandleSupplier.java:36)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.call(ExtensionMetadata.java:363)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.invoke(ExtensionMetadata.java:346)
	at org.jdbi.v3.core.extension.ExtensionFactoryDelegate.lambda$attach$0(ExtensionFactoryDelegate.java:118)
	at org.jdbi.v3.core.internal.OnDemandExtensions.invoke(OnDemandExtensions.java:86)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$2(OnDemandExtensions.java:72)
	at org.jdbi.v3.core.Jdbi.callWithExtension(Jdbi.java:557)
	at org.jdbi.v3.core.Jdbi.withExtension(Jdbi.java:544)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$3(OnDemandExtensions.java:72)
	at jdk.proxy2/jdk.proxy2.$Proxy27.findAll(Unknown Source)
	at dev.joseluisgs.productos.repository.ProductosRepositoryImpl.findAll(ProductosRepositoryImpl.kt:19)
	at dev.joseluisgs.productos.service.ProductosServiceImpl.getAll--HJRILA(ProductosServiceImpl.kt:29)
	at dev.joseluisgs.CarritoApp.run(Main.kt:134)
	at dev.joseluisgs.MainKt.main(Main.kt:36)
	at dev.joseluisgs.MainKt.main(Main.kt)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 41 more
Caused by: org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.SingleFileStore.lockFileChannel(SingleFileStore.java:143)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:117)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:81)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:286)
	... 35 more
2025-04-02 19:23:15.551662+02:00 database: flush
org.h2.message.DbException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:407)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:122)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:1546)
	at org.h2.mvstore.MVStore.panic(MVStore.java:371)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:291)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2035)
	at org.h2.mvstore.db.Store.<init>(Store.java:133)
	at org.h2.engine.Database.<init>(Database.java:326)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:344)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.Driver.connect(Driver.java:59)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:683)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:253)
	at org.jdbi.v3.core.Jdbi.lambda$create$0(Jdbi.java:122)
	at org.jdbi.v3.core.Jdbi.open(Jdbi.java:366)
	at org.jdbi.v3.core.LazyHandleSupplier.createHandle(LazyHandleSupplier.java:50)
	at org.jdbi.v3.core.internal.MemoizingSupplier.init(MemoizingSupplier.java:72)
	at org.jdbi.v3.core.internal.MemoizingSupplier.get(MemoizingSupplier.java:49)
	at org.jdbi.v3.core.LazyHandleSupplier.getHandle(LazyHandleSupplier.java:45)
	at org.jdbi.v3.sqlobject.statement.internal.CustomizingStatementHandler.invoke(CustomizingStatementHandler.java:186)
	at org.jdbi.v3.sqlobject.statement.internal.SqlQueryHandler.invoke(SqlQueryHandler.java:27)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.lambda$invoke$0(ExtensionMetadata.java:345)
	at org.jdbi.v3.core.AbstractHandleSupplier.invokeInContext(AbstractHandleSupplier.java:36)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.call(ExtensionMetadata.java:363)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.invoke(ExtensionMetadata.java:346)
	at org.jdbi.v3.core.extension.ExtensionFactoryDelegate.lambda$attach$0(ExtensionFactoryDelegate.java:118)
	at org.jdbi.v3.core.internal.OnDemandExtensions.invoke(OnDemandExtensions.java:86)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$2(OnDemandExtensions.java:72)
	at org.jdbi.v3.core.Jdbi.callWithExtension(Jdbi.java:557)
	at org.jdbi.v3.core.Jdbi.withExtension(Jdbi.java:544)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$3(OnDemandExtensions.java:72)
	at jdk.proxy2/jdk.proxy2.$Proxy27.findByCategoria(Unknown Source)
	at dev.joseluisgs.productos.repository.ProductosRepositoryImpl.findByCategoria(ProductosRepositoryImpl.kt:76)
	at dev.joseluisgs.productos.service.ProductosServiceImpl.getByCategoria-Zyo9ksc(ProductosServiceImpl.kt:116)
	at dev.joseluisgs.CarritoApp.run(Main.kt:138)
	at dev.joseluisgs.MainKt.main(Main.kt:36)
	at dev.joseluisgs.MainKt.main(Main.kt)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 41 more
Caused by: org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.SingleFileStore.lockFileChannel(SingleFileStore.java:143)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:117)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:81)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:286)
	... 35 more
2025-04-02 19:23:15.584658+02:00 database: flush
org.h2.message.DbException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:407)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:122)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:1546)
	at org.h2.mvstore.MVStore.panic(MVStore.java:371)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:291)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2035)
	at org.h2.mvstore.db.Store.<init>(Store.java:133)
	at org.h2.engine.Database.<init>(Database.java:326)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:344)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.Driver.connect(Driver.java:59)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:683)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:253)
	at org.jdbi.v3.core.Jdbi.lambda$create$0(Jdbi.java:122)
	at org.jdbi.v3.core.Jdbi.open(Jdbi.java:366)
	at org.jdbi.v3.core.LazyHandleSupplier.createHandle(LazyHandleSupplier.java:50)
	at org.jdbi.v3.core.internal.MemoizingSupplier.init(MemoizingSupplier.java:72)
	at org.jdbi.v3.core.internal.MemoizingSupplier.get(MemoizingSupplier.java:49)
	at org.jdbi.v3.core.LazyHandleSupplier.getHandle(LazyHandleSupplier.java:45)
	at org.jdbi.v3.sqlobject.statement.internal.CustomizingStatementHandler.invoke(CustomizingStatementHandler.java:186)
	at org.jdbi.v3.sqlobject.statement.internal.SqlQueryHandler.invoke(SqlQueryHandler.java:27)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.lambda$invoke$0(ExtensionMetadata.java:345)
	at org.jdbi.v3.core.AbstractHandleSupplier.invokeInContext(AbstractHandleSupplier.java:36)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.call(ExtensionMetadata.java:363)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.invoke(ExtensionMetadata.java:346)
	at org.jdbi.v3.core.extension.ExtensionFactoryDelegate.lambda$attach$0(ExtensionFactoryDelegate.java:118)
	at org.jdbi.v3.core.internal.OnDemandExtensions.invoke(OnDemandExtensions.java:86)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$2(OnDemandExtensions.java:72)
	at org.jdbi.v3.core.Jdbi.callWithExtension(Jdbi.java:557)
	at org.jdbi.v3.core.Jdbi.withExtension(Jdbi.java:544)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$3(OnDemandExtensions.java:72)
	at jdk.proxy2/jdk.proxy2.$Proxy39.findAll(Unknown Source)
	at dev.joseluisgs.clientes.repository.ClientesRepositoryImpl.findAll(ClientesRepositoryImpl.kt:18)
	at dev.joseluisgs.clientes.service.ClientesServiceImpl.getAll--HJRILA(ClientesServiceImpl.kt:29)
	at dev.joseluisgs.CarritoApp.run(Main.kt:143)
	at dev.joseluisgs.MainKt.main(Main.kt:36)
	at dev.joseluisgs.MainKt.main(Main.kt)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 41 more
Caused by: org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.SingleFileStore.lockFileChannel(SingleFileStore.java:143)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:117)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:81)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:286)
	... 35 more
2025-04-02 19:23:15.599659+02:00 database: flush
org.h2.message.DbException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:407)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:122)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:1546)
	at org.h2.mvstore.MVStore.panic(MVStore.java:371)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:291)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2035)
	at org.h2.mvstore.db.Store.<init>(Store.java:133)
	at org.h2.engine.Database.<init>(Database.java:326)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:344)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.Driver.connect(Driver.java:59)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:683)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:253)
	at org.jdbi.v3.core.Jdbi.lambda$create$0(Jdbi.java:122)
	at org.jdbi.v3.core.Jdbi.open(Jdbi.java:366)
	at org.jdbi.v3.core.LazyHandleSupplier.createHandle(LazyHandleSupplier.java:50)
	at org.jdbi.v3.core.internal.MemoizingSupplier.init(MemoizingSupplier.java:72)
	at org.jdbi.v3.core.internal.MemoizingSupplier.get(MemoizingSupplier.java:49)
	at org.jdbi.v3.core.LazyHandleSupplier.getHandle(LazyHandleSupplier.java:45)
	at org.jdbi.v3.sqlobject.statement.internal.CustomizingStatementHandler.invoke(CustomizingStatementHandler.java:186)
	at org.jdbi.v3.sqlobject.statement.internal.SqlQueryHandler.invoke(SqlQueryHandler.java:27)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.lambda$invoke$0(ExtensionMetadata.java:345)
	at org.jdbi.v3.core.AbstractHandleSupplier.invokeInContext(AbstractHandleSupplier.java:36)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.call(ExtensionMetadata.java:363)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.invoke(ExtensionMetadata.java:346)
	at org.jdbi.v3.core.extension.ExtensionFactoryDelegate.lambda$attach$0(ExtensionFactoryDelegate.java:118)
	at org.jdbi.v3.core.internal.OnDemandExtensions.invoke(OnDemandExtensions.java:86)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$2(OnDemandExtensions.java:72)
	at org.jdbi.v3.core.Jdbi.callWithExtension(Jdbi.java:557)
	at org.jdbi.v3.core.Jdbi.withExtension(Jdbi.java:544)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$3(OnDemandExtensions.java:72)
	at jdk.proxy2/jdk.proxy2.$Proxy39.findById(Unknown Source)
	at dev.joseluisgs.clientes.repository.ClientesRepositoryImpl.findById(ClientesRepositoryImpl.kt:23)
	at dev.joseluisgs.clientes.repository.ClientesRepositoryImpl.findById(ClientesRepositoryImpl.kt:11)
	at dev.joseluisgs.clientes.service.ClientesServiceImpl.getById-Zyo9ksc(ClientesServiceImpl.kt:42)
	at dev.joseluisgs.CarritoApp.run(Main.kt:147)
	at dev.joseluisgs.MainKt.main(Main.kt:36)
	at dev.joseluisgs.MainKt.main(Main.kt)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 42 more
Caused by: org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.SingleFileStore.lockFileChannel(SingleFileStore.java:143)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:117)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:81)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:286)
	... 36 more
2025-04-02 19:23:15.610661+02:00 database: flush
org.h2.message.DbException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:407)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:122)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:1546)
	at org.h2.mvstore.MVStore.panic(MVStore.java:371)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:291)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2035)
	at org.h2.mvstore.db.Store.<init>(Store.java:133)
	at org.h2.engine.Database.<init>(Database.java:326)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:344)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.Driver.connect(Driver.java:59)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:683)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:253)
	at org.jdbi.v3.core.Jdbi.lambda$create$0(Jdbi.java:122)
	at org.jdbi.v3.core.Jdbi.open(Jdbi.java:366)
	at org.jdbi.v3.core.LazyHandleSupplier.createHandle(LazyHandleSupplier.java:50)
	at org.jdbi.v3.core.internal.MemoizingSupplier.init(MemoizingSupplier.java:72)
	at org.jdbi.v3.core.internal.MemoizingSupplier.get(MemoizingSupplier.java:49)
	at org.jdbi.v3.core.LazyHandleSupplier.getHandle(LazyHandleSupplier.java:45)
	at org.jdbi.v3.sqlobject.statement.internal.CustomizingStatementHandler.invoke(CustomizingStatementHandler.java:186)
	at org.jdbi.v3.sqlobject.statement.internal.SqlQueryHandler.invoke(SqlQueryHandler.java:27)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.lambda$invoke$0(ExtensionMetadata.java:345)
	at org.jdbi.v3.core.AbstractHandleSupplier.invokeInContext(AbstractHandleSupplier.java:36)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.call(ExtensionMetadata.java:363)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.invoke(ExtensionMetadata.java:346)
	at org.jdbi.v3.core.extension.ExtensionFactoryDelegate.lambda$attach$0(ExtensionFactoryDelegate.java:118)
	at org.jdbi.v3.core.internal.OnDemandExtensions.invoke(OnDemandExtensions.java:86)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$2(OnDemandExtensions.java:72)
	at org.jdbi.v3.core.Jdbi.callWithExtension(Jdbi.java:557)
	at org.jdbi.v3.core.Jdbi.withExtension(Jdbi.java:544)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$3(OnDemandExtensions.java:72)
	at jdk.proxy2/jdk.proxy2.$Proxy39.findById(Unknown Source)
	at dev.joseluisgs.clientes.repository.ClientesRepositoryImpl.findById(ClientesRepositoryImpl.kt:23)
	at dev.joseluisgs.clientes.repository.ClientesRepositoryImpl.findById(ClientesRepositoryImpl.kt:11)
	at dev.joseluisgs.clientes.service.ClientesServiceImpl.getById-Zyo9ksc(ClientesServiceImpl.kt:42)
	at dev.joseluisgs.CarritoApp.run(Main.kt:153)
	at dev.joseluisgs.MainKt.main(Main.kt:36)
	at dev.joseluisgs.MainKt.main(Main.kt)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 42 more
Caused by: org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.SingleFileStore.lockFileChannel(SingleFileStore.java:143)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:117)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:81)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:286)
	... 36 more
2025-04-02 19:23:15.641658+02:00 database: flush
org.h2.message.DbException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:407)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:122)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:1546)
	at org.h2.mvstore.MVStore.panic(MVStore.java:371)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:291)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2035)
	at org.h2.mvstore.db.Store.<init>(Store.java:133)
	at org.h2.engine.Database.<init>(Database.java:326)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:344)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.Driver.connect(Driver.java:59)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:683)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:253)
	at org.jdbi.v3.core.Jdbi.lambda$create$0(Jdbi.java:122)
	at org.jdbi.v3.core.Jdbi.open(Jdbi.java:366)
	at org.jdbi.v3.core.LazyHandleSupplier.createHandle(LazyHandleSupplier.java:50)
	at org.jdbi.v3.core.internal.MemoizingSupplier.init(MemoizingSupplier.java:72)
	at org.jdbi.v3.core.internal.MemoizingSupplier.get(MemoizingSupplier.java:49)
	at org.jdbi.v3.core.LazyHandleSupplier.getHandle(LazyHandleSupplier.java:45)
	at org.jdbi.v3.sqlobject.statement.internal.CustomizingStatementHandler.invoke(CustomizingStatementHandler.java:186)
	at org.jdbi.v3.sqlobject.statement.internal.SqlQueryHandler.invoke(SqlQueryHandler.java:27)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.lambda$invoke$0(ExtensionMetadata.java:345)
	at org.jdbi.v3.core.AbstractHandleSupplier.invokeInContext(AbstractHandleSupplier.java:36)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.call(ExtensionMetadata.java:363)
	at org.jdbi.v3.core.extension.ExtensionMetadata$ExtensionHandlerInvoker.invoke(ExtensionMetadata.java:346)
	at org.jdbi.v3.core.extension.ExtensionFactoryDelegate.lambda$attach$0(ExtensionFactoryDelegate.java:118)
	at org.jdbi.v3.core.internal.OnDemandExtensions.invoke(OnDemandExtensions.java:86)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$2(OnDemandExtensions.java:72)
	at org.jdbi.v3.core.Jdbi.callWithExtension(Jdbi.java:557)
	at org.jdbi.v3.core.Jdbi.withExtension(Jdbi.java:544)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$3(OnDemandExtensions.java:72)
	at jdk.proxy2/jdk.proxy2.$Proxy40.findById(Unknown Source)
	at dev.joseluisgs.ventas.repository.VentasRepositoryImpl.findById(VentasRepositoryImpl.kt:63)
	at dev.joseluisgs.ventas.repository.VentasRepositoryImpl.findById(VentasRepositoryImpl.kt:16)
	at dev.joseluisgs.ventas.service.VentasServiceImpl.getById-Zyo9ksc(VentasServiceImpl.kt:34)
	at dev.joseluisgs.CarritoApp.run(Main.kt:163)
	at dev.joseluisgs.MainKt.main(Main.kt:36)
	at dev.joseluisgs.MainKt.main(Main.kt)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: Error General : "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]" [50000-232]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 42 more
Caused by: org.h2.mvstore.MVStoreException: The file is locked: C:/Users/joseluisgs/Projects/CarroCompra/ventas.mv.db [2.3.232/7]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.SingleFileStore.lockFileChannel(SingleFileStore.java:143)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:117)
	at org.h2.mvstore.SingleFileStore.open(SingleFileStore.java:81)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:286)
	... 36 more
